LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/CommonAPI/LoRaCloudToDeviceMessage.cs

```diff
@@ -5,13 +5,22 @@ namespace LoRaTools.CommonAPI
 {
     using System.Collections.Generic;
     using LoRaWan;
+    using Newtonsoft.Json;
 
     /// <summary>
     /// Defines the contract for a LoRa cloud to device message.
     /// </summary>
     public class LoRaCloudToDeviceMessage : ILoRaCloudToDeviceMessage
     {
-        public string DevEUI { get; set; }
+        [JsonIgnore]
+        public DevEui? DevEUI { get; set; }
+
+        [JsonProperty("DevEUI")]
+        public string DevEuiString
+        {
+            get => DevEUI?.ToString("N", null);
+            set => DevEUI = string.IsNullOrEmpty(value) ? null : DevEui.Parse(value);
```

Comment:
Some of the serialization does plain .Parse, others do use TryParse and others do some minor validation (like this one). Some of them may allow null and that's fine (would be good to check again), but here, it seems we should be using TryParse as well?

Reply:
I think there's a minor difference here that might be useful - if the `value` is not `null` or empty, we enforce that it is a parseable `DevEui`. In case of something invalid (e.g. `"1234"`) we fail faster compared to just using `TryParse`

Reply:
My comment was more around consistency in handling that assignment in the other types.

Reply:
As discussed offline, will resolve this. There is a related comment which I will leave open for now.